HPCC-33465 Remove legacy hextoll function and use strtoll instead

streeterd opened the pull request:
Remove hextoll function and use strtoll instead

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Cloud-compatibility
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

streeterd committed:
HPCC-33465 Remove legacy hextoll function and use strtoll instead

Remove hextoll function and use strtoll instead

Signed-off-by: Dave Streeter <dave.streeter@lexisnexisrisk.com>

jakesmith reviewed (commented):
@streeterd - please go ahead and squash. Also, in this case, I think worth changing the JIRA title and commit heading to "HPCC-33465 Remove legacy hextoll function and use strtoll instead"

streeterd commented:
Squashed and renamed




Thanks and Regards


Dave Streeter

Senior Software Engineer
LexisNexis Risk Solutions | HPCC Systems | RELX Group
Tel: +44 7972 845 177


________________________________
From: Jake Smith ***@***.***>
Sent: Wednesday, February 19, 2025 10:37 AM
To: hpcc-systems/HPCC-Platform ***@***.***>
Cc: Streeter, Dave (RIS-HBE) ***@***.***>; Mention ***@***.***>
Subject: Re: [hpcc-systems/HPCC-Platform] HPCC-33465 Fix minor coverity issue in Dali (PR #19536)


*** External email: use caution ***



@jakesmith commented on this pull request.

@streeterd<https://github.com/streeterd> - please go ahead and squash. Also, in this case, I think worth changing the JIRA title and commit heading to "HPCC-33465 Remove legacy hextoll function and use strtoll instead"

—
Reply to this email directly, view it on GitHub<https://github.com/hpcc-systems/HPCC-Platform/pull/19536#pullrequestreview-2626351205>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BPK7I53AGTWRTML47LKOHA32QRNF5AVCNFSM6AAAAABXNYMBDWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDMMRWGM2TCMRQGU>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________
The information contained in this e-mail message is intended only for the personal and confidential use of the recipient(s) named above. This message may be an attorney-client communication and/or work product and as such is privileged and confidential. If the reader of this message is not the intended recipient or an agent responsible for delivering it to the intended recipient, you are hereby notified that you have received this document in error and that any review, dissemination, distribution, or copying of this message is strictly prohibited. If you have received this communication in error, please notify us immediately by e-mail, and delete the original message.

jakesmith reviewed (approved):
@streeterd - looks good.
